app/controllers/mixins/generic_show_mixin.rb

```diff
@@ -66,26 +66,35 @@ def display_nested_list(display)
       respond_to?(nested_list_method_name(display).to_sym) ? nested_list_call(display) : display_nested_generic(display)
     end
 
+    def prepare_breadcrumb(title)
+      title = title.is_a?(Hash) ? _("(All %{resource_name})") % {:resource_name => ui_lookup(:tables => title[:table])} : title
+      bc_name = "#{@record.name} #{title}"
+      url     = "/#{self.class.table_name}/show/#{@record.id}?display=#{@display}"
+      drop_breadcrumb(:name => bc_name, :url => url)
+    end
+
+
     def display_nested_generic(display)
-      nested_list(display, display.camelize.singularize.constantize)
+      nested_list({:table => display}, display.camelize.singularize.constantize)
     end
 
     def display_instances
-      nested_list("vm_cloud", ManageIQ::Providers::CloudManager::Vm)
+      nested_list({:table => "vm_cloud"}, ManageIQ::Providers::CloudManager::Vm)
     end
 
     def display_images
-      nested_list("template_cloud", ManageIQ::Providers::CloudManager::Template)
+      nested_list({:table => "template_cloud"}, ManageIQ::Providers::CloudManager::Template)
     end
 
-    def nested_list(table_name, model)
-      title = ui_lookup(:tables => table_name)
-      drop_breadcrumb(:name => _("%{name} (Summary)") % {:name => @record.name},
-                      :url  => "/#{self.class.table_name}/show/#{@record.id}")
-      drop_breadcrumb(:name => _("%{name} (All %{title})") % {:name => @record.name, :title => title},
-                      :url  => "/#{self.class.table_name}/show/#{@record.id}?display=#{@display}")
-      @view, @pages = get_view(model, :parent => @record) # Get the records (into a view) and the paginator
-      @showtype     = @display
+
+    def nested_list(table_name, model, association=nil)
```

Comment:
`get_view`'s options default to `{}` so maybe also `nested_list` could have `options` that would default to `{}` and would be passed on to `get_view`.

That would remove the need for the `if` statement below.